CMakeLists.txt

```diff
@@ -30,7 +36,7 @@ target_include_directories(nlohmann_json
 ################################################################################
 if (NOT EXISTS "${CMAKE_BINARY_DIR}/External/optional-lite/nonstd/optional.hpp")
     file(DOWNLOAD
-        "https://github.com/martinmoene/optional-lite/releases/download/v3.0.0/optional.hpp"
+        "https://raw.githubusercontent.com/martinmoene/optional-lite/master/include/nonstd/optional.hpp"
```

Comment:
Maybe change this from `master` to the specific commit (that is, use the URL https://raw.githubusercontent.com/martinmoene/optional-lite/e240fdaaa4071a980926dea4ed5b610aa8fbc538/include/nonstd/optional.hpp)?
That way, all users get the same version of `optional-lite` when they build `netket`.